tests/ui/never_type/lint-breaking-2024-assign-underscore.fixed

```diff
@@ -0,0 +1,17 @@
+//@ run-rustfix
+
+#![allow(unused)]
+#![deny(dependency_on_unit_never_type_fallback)]
+
+fn foo<T: Default>() -> Result<T, ()> {
+    Err(())
+}
+
+fn test() -> Result<(), ()> {
+    //~^ ERROR this function depends on never type fallback being `()`
+    //~| WARN this was previously accepted by the compiler but is being phased out
+    _ = foo::<()>()?;
```

Comment:
imo the better suggestion would be to `() = foo()?;`, at least it's less noisy than `::<()>` 🤔

Reply:
that is impossible to suggest given how this desugars into two let statements

Reply:
or at least i don't want to look into it

Reply:
welp :(